app/src/main/java/org/android/go/sopt/ui/home/HomeAdapter.kt

```diff
@@ -0,0 +1,119 @@
+import org.android.go.sopt.databinding.ItemTextBinding
+import org.android.go.sopt.model.FakeGithubInfoItem
+
+class HomeAdapter() :
+    ListAdapter<FakeGithubInfoItem, RecyclerView.ViewHolder>(
+        FakeGithubDiffCallback()
+    ) {
+
+    private lateinit var selectionTracker: SelectionTracker<Long>
+
+    init {
+        setHasStableIds(true)
+    }
+
+    override fun getItemId(position: Int): Long = position.toLong()
+
+    override fun getItemViewType(position: Int): Int =
+        when {
+            getItem(position)?.title != null -> VIEW_TYPE_TITLE
+            else -> VIEW_TYPE_ITEM
+        }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder =
+        when (viewType) {
+            VIEW_TYPE_TITLE -> TitleViewHolder(
+                ItemTextBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+            )
+            VIEW_TYPE_ITEM -> HomeViewHolder(
+                ItemGithubInfoBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+            )
+            else -> throw IllegalArgumentException("Invalid view type")
+        }
+
+    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+        val content = currentList[position]
+        when (holder) {
+            is HomeViewHolder -> holder.bind(content, position)
+            is TitleViewHolder -> holder.bind(content)
+        }
+    }
+
+
+    fun setSelectionTracker(selectionTracker: SelectionTracker<Long>) {
+        this.selectionTracker = selectionTracker
+    }
+
+    /**
+     * ItemDetailsLookup
+     * 사용자가 선택한 아이템의 위치,정보를 제공
+     * getItemDetails메서드 제공 인터페이스
+     * MotionEvent를 기반하여 선택된 내용을 ViewHolder에 매핑
+     * */
+    class HomeLookUp(private val recyclerView: RecyclerView) : ItemDetailsLookup<Long>() {
+        override fun getItemDetails(event: MotionEvent): ItemDetails<Long>? {
+            val view = recyclerView.findChildViewUnder(event.x, event.y)
+            return view?.let {
+                val viewHolder = recyclerView.getChildViewHolder(it) as? HomeViewHolder
+                viewHolder?.getItemDetails()
+            }
+        }
+    }
+
+    inner class HomeViewHolder(private val binding: ItemGithubInfoBinding) :
+        RecyclerView.ViewHolder(binding.root) {
+        fun bind(fakeGithubInfoItem: FakeGithubInfoItem, itemPosition: Int) = with(binding) {
+            fakeinfo = fakeGithubInfoItem
+            root.setOnClickListener {
+                selectionTracker?.select(itemPosition.toLong())
+            }
+
+            loRvitem.isActivated = selectionTracker?.isSelected(itemPosition.toLong()) ?: false
+            executePendingBindings()
+        }
+
+        fun getItemDetails(): ItemDetailsLookup.ItemDetails<Long> =
+            object : ItemDetailsLookup.ItemDetails<Long>() {
+                override fun getPosition(): Int = adapterPosition
+                override fun getSelectionKey(): Long? = itemId
```

Comment:
```suggestion
                override fun getSelectionKey(): Long = itemId
```

itemId 또한 non-null 타입을 반환하기 때문에 `?`는 없어도 될 것 같습니다!